### PR TITLE
[Task] YD-2315: Support v2.4.0 of the SDK

### PR DESCRIPTION
**Pull request by asklausen**
## Purpose
Support v2.4.0 of the SDK.

## External References (e.g. Jira / Zeplin / Confluence)
[YD-2315](https://lampkicking.atlassian.net/browse/YD-2315)

## Approach
- Modify properties and capabilities of the project's target relative to NFC (according to [this](https://github.com/getyoti/yoti-doc-scan-ios/blob/master/README.md))
- Updated the `podfile` with `use_frameworks` and a deployment target of 11.0

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -38,8 +38,8 @@ android {
 dependencies {
     //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:${safeExtGet('reactNativeVersion', '+')}"
-    implementation 'com.yoti.mobile.android.sdk:yoti-sdk-doc-scan:2.3.2'
-    implementation 'com.yoti.mobile.android.sdk:yoti-sdk-liveness-zoom:2.3.2'
+    implementation 'com.yoti.mobile.android.sdk:yoti-sdk-doc-scan:2.4.0'
+    implementation 'com.yoti.mobile.android.sdk:yoti-sdk-liveness-zoom:2.4.0'
```
Comment (mariasenosiain) [inline android/build.gradle:43]: Oh! I forget to told you that we need to include another library for Supplementary document implementation 🙈 
Just add this lie below `liveness-zoom` one:
```
implementation 'com.yoti.mobile.android.sdk:yoti-sdk-doc-scan-sup:2.4.0'
```

File: react-native-yoti-doc-scan.podspec [hunk 2]
```diff
@@ -2,21 +2,20 @@ require "json"
...
+  spec.name                   = "react-native-yoti-doc-scan"
+  spec.version                = package["version"]
+  spec.summary                = package["description"]
+  spec.description            = package["description"]
+  spec.license                = package["license"]
+  spec.author                 = package["author"]
+  spec.homepage               = "https://github.com/getyoti/yoti-doc-scan-react-native"
+  spec.source                 = { :git => "https://github.com/getyoti/yoti-doc-scan-react-native.git", :tag => "#{spec.version}" }
+  spec.source_files           = "ios/**/*.{h,m}""
+  spec.platform               = :ios, "12.0"
```
Comment (mariasenosiain) [inline react-native-yoti-doc-scan.podspec]: A doubt about this (good to know it if I need to update podspec file next release): Why it cannot be 11.0? I've seen that you added `spec.ios.deployment_target  = "11.0"`, but not sure what's the difference and why now both are needed :happy:

Comment (asklausen) [reply react-native-yoti-doc-scan.podspec]: `platform` refers to the min. version of Xcode, while `deployment_target` refers to the min. version of iOS.
If only `platform` is set, then CocoaPods will apply that min. version to `deployment_target` as well.

To answer your specific question on why `platform` can't be 11 or 11.4 as it was previously set to: iOS v2.4.0 simply won't work on those versions of Xcode given their associated versions of the Swift compiler. iOS v2.4.0 will thus only work with the Swift compiler associated to Xcode 12. And in order for the `deployment_target` to then not be set to 12 as well, we need to add this to the spec as 11.

Comment (mariasenosiain) [reply react-native-yoti-doc-scan.podspec]: Oh! I see! Thanks for the explanation :)

File: android/build.gradle (hunk 1 above)
Comment (asklausen) [reply android/build.gradle:43]: Added it in between just to keep the order of id doc, sup doc and liveness.

**Review by mariasenosiain — APPROVED** [review]

File: react-native-yoti-doc-scan.podspec (hunk 2 above)
Comment (asklausen) [reply react-native-yoti-doc-scan.podspec]: @mariasenosiain This proved to not work as how I thought. When verifying, I noticed that the target set relative to `platform` is in fact the `deployment_target`, so I've modified the `podspec` and `podfile` to reflect that (I've also adjusted it on our iOS repository, so no issues there).

**Review by AdrienVioletYoti — APPROVED** [review]

**Comment by mariasenosiain** [comment]
B&T-ing

**Comment by mariasenosiain** [comment]
B&T passed. 
- LG-K8 - Android 6.0
- iPhone X - iOS 13.7

Tested in both platforms:

- Complete ID Doc. Passport + NFC flow.
- Complete liveness flow.
- Complete Sup. Document flow.

**Comment by Zzhelyaz** [comment]
Verified.

**Pull request merged**